deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1304 @@
+    enabledLogType[logTypes.Error] = false; // For development purpose
+
+    function logFunc(logType, argsArray) {
+      if (enabledLogType[logType]) {
+        var printStr = '';
+        for (var index in argsArray) {
+          var obj = argsArray[index];
+          if (obj != undefined) {
+            printStr += (typeof obj != 'string') ? JSON.stringify(obj) : obj;
+            printStr += ': ';
+          }
+        }
+        chakraDebug.log(printStr);
+      }
+    }
+
+    return {
+      LogAPI: function(msg, obj) {
+        logFunc(logTypes.API, arguments);
+      },
+      LogOutput: function(msg, obj) {
+        logFunc(logTypes.Output, arguments);
+      },
+      LogInput: function(msg, obj) {
+        logFunc(logTypes.Input, arguments);
+      },
+      LogError: function(msg, obj) {
+        logFunc(logTypes.Error, arguments);
+      }
+    };
+  })();
+
+  function CallHostFunctionNoLog(fn) {
+    var args = [].slice.call(arguments, 1);
+    var start = new Date();
+    var result = fn.apply(undefined, args);
+    var end = new Date();
+    Logger.LogAPI(`${fn.name}(${args}):${end - start}`);
+    return result;
+  }
+
+  function CallHostFunction(fn) {
+    var args = [].slice.call(arguments, 1);
+    var start = new Date();
+    var result = fn.apply(undefined, args);
+    var end = new Date();
+    Logger.LogAPI(`${fn.name}(${args}):${end - start}`, result);
+    return result;
+  }
+
+  if (!globalObject.Debug) {
+    Object.defineProperty(globalObject, 'Debug', {
+      value: {}, enumerable: false, configurable: false, writable: false
+    });
+  }
+
+  var Debug = globalObject.Debug;
+
+  Debug.ScriptBreakPointType = {
+    ScriptId: 0,
+    ScriptName: 1,
+    ScriptRegExp: 2
+  };
+
+  Debug.setListener = function(fn) { };
+
+  Debug.setBreakPoint = function(func, line, column) {
+    Logger.LogAPI(`Debug.setBreakPoint(${func},${line},${column})`);
+    var funcInfo = CallHostFunction(chakraDebug.JsDiagGetFunctionPosition,
+                                     func);
+    var bpId = -1;
+
+    if (funcInfo.scriptId >= 0) {
+      var scriptObj = DebugManager.ScriptsManager.GetScript(funcInfo.scriptId);
+      var bpLine = funcInfo.firstStatementLine + line;
+      var bpColumn = funcInfo.firstStatementColumn + column;
+      var v8Breakpoint = new V8Breakpoint(Debug.ScriptBreakPointType.ScriptId,
+                              funcInfo.scriptId, scriptObj, bpLine, bpColumn);
+      if (v8Breakpoint.Set()) {
+        bpId = v8Breakpoint.Id();
+        DebugManager.BreakpointManager.AddBreakpoint(v8Breakpoint);
+      }
+    }
+
+    return bpId;
+  };
+
+  function AddChildrens(obj) {
+    if ('display' in obj) {
+      if (!('value' in obj) || !isFinite(obj['value']) ||
+        obj['value'] == null || obj['value'] == undefined) {
+        obj['value'] = obj['display'];
+      }
+      if (!('text' in obj)) {
+        obj['text'] = obj['display'];
+      }
+      if (!('className' in obj)) {
```

Comment:
If you look at JsrtDebugUtils::AddPropertyType, we always add "type" but className in  only for few cases. This adding of className as type is to satisfy VSCode.
hasOwnProperty will do but it seems most of the place we use in to check for property. 

---
In reply to: [81491092](https://github.com/nodejs/node-chakracore/pull/126#discussion_r81491092) [](ancestors = 81491092)